test/extended/router/router.go

```diff
@@ -97,21 +104,24 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
 })
 
 func waitForFirstRouterEndpointIP(oc *exutil.CLI) (string, error) {
-	_, ns, err := exutil.GetRouterPodTemplate(oc)
-	if err != nil {
-		return "", err
-	}
-
+	oc.KubeFramework().Namespace.Name = "openshift-ingress"
 	// wait for at least one router endpoint to be up router endpoints to show up
 	var host string
-	err = wait.PollImmediate(2*time.Second, 120*time.Second, func() (bool, error) {
-		epts, err := oc.AdminKubeClient().CoreV1().Endpoints(ns).Get("router", metav1.GetOptions{})
-		o.Expect(err).NotTo(o.HaveOccurred())
-		if len(epts.Subsets) == 0 || len(epts.Subsets[0].Addresses) == 0 {
-			return false, nil
+	err := wait.PollImmediate(2*time.Second, 120*time.Second, func() (bool, error) {
+		currlabel := "router=router"
+		for _, ns := range []string{"default", "openshift-ingress", "tectonic-ingress"} {
+			//Search the router by label
+			if ns == "openshift-ingress" {
+				currlabel = "app=router"
```

Comment:
Should break early and should probably be looking for `router=router-default` as well

Reply:
Less important than the short circuiting is the fact that we can have multiple routers, and the tests should be by default targeting *just the default router*.